node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -74,23 +72,45 @@ class DataMessageHandler(callbacks: SpvNodeCallbacks, chainHandler: ChainApi)(
         val headers = headersMsg.headers
         val chainApiF = chainHandler.processHeaders(headers)
 
-        chainApiF.map { newApi =>
-          val lastHeader = headers.last
-          val lastHash = lastHeader.hash
-          newApi.getBlockCount.map { count =>
-            logger.trace(
-              s"Processed headers, most recent has height=$count and hash=$lastHash.")
+        chainApiF
+          .map { newApi =>
+            if (headers.nonEmpty) {
```

Comment:
As we talked about on our call, this is going to conflict with #655 

https://github.com/bitcoin-s/bitcoin-s/pull/655/files#diff-a3d17dc45d049885d0cd078caca38e9cR79

I guess the design question here is: If we receive a headers message, and it appears we are not synced, is it the user's responsibility to call `SpvNode.sync()` or is it `DataMessageHandler`'s responsibility?

With things like `sendheaders`, we should stay in sync once IBD is done. My experimental branch seems to do IBD ok, although I can't recall where the next send headers message is coming from https://github.com/Christewart/bitcoin-s-core/tree/2019-07-31-recv-funds-rebase-655

https://github.com/Christewart/bitcoin-s-core/blob/2019-07-31-recv-funds-rebase-655/node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala#L89

Reply:
I can't really imagine a scenario where it's intentional to _not stay in sync_. So it would always make sense for `DataMessageHandler` to request the newest information if we're behind

Reply:
So what heuristic would you use? If the headers message is of size 2000? That is the max headers size https://bitcoin.org/en/developer-reference#getheaders

If we receive less headers, that probably indicates we are at the end of sync, right? Otherwise I think we end up in a infinite loop of request the last header we've seen. 

i.e. 

```
1. sendheaders(lasthash)
2. receive headermessage(Vector(lastheader))
3. process to data message handler
4. sendheaders(same-last-hash)
```
and we keep doing that until a new block comes in